### PR TITLE
UIREC-291: add dropdown action buttons

### PR DESCRIPTION
**Pull request by alisher-epam**
## Purpose

[UIREC-291](https://issues.folio.org/browse/UIREC-291) - Combine piece action buttons into action menu
## Approach

https://github.com/folio-org/ui-receiving/assets/116072773/fef07c4d-c90e-4204-b6b0-806adc3dea64 

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by OleksandrHladchenko1 — APPROVED** [review]

**Review by UladzislauKutarkin — APPROVED** [review]

File: src/TitleDetails/AddPieceModal/ModalActionButtons/ModalActionButtons.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  });
+  const saveButtonLabelId = isCreateAnother ? 'stripes-core.button.save' : 'ui-receiving.piece.actions.saveAndClose';
+
+  if (actionMenus.length === 0) {
+    return (
+      <Button
+        buttonStyle="primary"
+        data-test-add-piece-save
+        disabled={disabled}
+        onClick={onSave}
```
Comment (usavkov-epam) [inline src/TitleDetails/AddPieceModal/ModalActionButtons/ModalActionButtons.js:41]: Add please `marginBottom0` here

File: src/TitleDetails/AddPieceModal/ModalActionButtons/ModalActionButtons.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    );
+  }
+
+  return (
+    <ButtonGroup>
+      <Button
+        buttonStyle="primary"
+        data-test-add-piece-save
+        disabled={disabled}
+        onClick={onSave}
```
Comment (usavkov-epam) [inline src/TitleDetails/AddPieceModal/ModalActionButtons/ModalActionButtons.js:54]: `marginBottom0` plz

File: src/TitleDetails/AddPieceModal/ModalActionButtons/ModalActionButtons.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      >
+        <FormattedMessage id={saveButtonLabelId} />
+      </Button>
+      <Dropdown buttonProps={{ buttonStyle: 'primary', buttonClass: css.dropdownButton }}>
```
Comment (usavkov-epam) [inline src/TitleDetails/AddPieceModal/ModalActionButtons/ModalActionButtons.js:58]:
```
{
  buttonStyle: 'primary',
  buttonClass: css.dropdownButton,
  marginBottom0: true,
}
```

File: src/TitleDetails/AddPieceModal/ModalActionButtons/utils.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { PIECE_ACTIONS, PIECE_ACTIONS_BY_STATUS } from './constants';
+
+export const getPieceActionMenus = ({ status, disabled, ...rest }) => {
```
Comment (usavkov-epam) [inline src/TitleDetails/AddPieceModal/ModalActionButtons/utils.js:3]: is there any reason to destructure `disabled`?

File: src/TitleDetails/AddPieceModal/ModalActionButtons/constants.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+      buttonStyle="dropdownItem"
+      onClick={onReceive}
+    >
+      <Icon icon="receive">
+        <FormattedMessage id="ui-receiving.piece.action.button.quickReceive" />
+      </Icon>
+    </Button>
+  ),
+  saveAndCreate: (
+    <Button disabled={disabled} buttonStyle="dropdownItem">
```
Comment (usavkov-epam) [inline src/TitleDetails/AddPieceModal/ModalActionButtons/constants.js:93]: "Save and create another" stoped working here?

**Review by usavkov-epam — CHANGES_REQUESTED** [review]

**Comment by usavkov-epam** [comment]
It seems that "Save and create another" not working after changes

**Comment by usavkov-epam** [comment]
https://github.com/folio-org/ui-receiving/assets/88109087/6caa5884-acbf-4bb4-87e6-6f1db979ebe6

**Comment by usavkov-epam** [comment]
And I don't think we need "Delete" action for a new piece form

**Comment by usavkov-epam** [comment]
Since we have the new action for "Save & create another", "Save&close" button should always save a piece and close the modal. Currently after clicking "Save & create another", `isCreateAnother` state stays the same for the next piece modal: "save&close" btn change label (it must not) and after clicking just "save" it open modal again (we have separate action for that).

https://github.com/folio-org/ui-receiving/assets/88109087/c88cc587-987a-4e1a-a38b-7f67c528ba44

**Comment by alisher-epam** [comment]
@usavkov-epam here is the screen record for fixing

https://github.com/folio-org/ui-receiving/assets/116072773/7b33ed98-ea8c-40b1-9896-3641b312941c

File: src/TitleDetails/AddPieceModal/AddPieceModal.js [hunk 6]
```diff
@@ -128,10 +128,10 @@ const AddPieceModal = ({
     return handleSubmit(e);
   }, [checkHoldingAbandonment, getState, handleSubmit, id, initialHoldingId, toggleDeleteHoldingsConfirmation]);
 
-  const onCreateAnotherPiece = useCallback((e) => {
+  const onCreateAnotherPiece = useCallback(() => {
     change('isCreateAnother', true);
-    onSave(e);
-  }, [change, onSave]);
+    handleSubmit();
```
[the remark this answers is not in the document]
Comment (usavkov-epam) [reply src/TitleDetails/AddPieceModal/AddPieceModal.js:133]: I've checked locally - it's not

File: src/TitleDetails/AddPieceModal/ModalActionButtons/ModalActionButtons.css [hunk 7]
```diff
@@ -0,0 +1,11 @@
+.dropdownButton {
+  border-bottom-left-radius: 0;
+  border-top-left-radius: 0;
+  margin-left: -1px;
+
+  &::before, &:focus::before {
```
Comment (usavkov-epam) [inline src/TitleDetails/AddPieceModal/ModalActionButtons/ModalActionButtons.css:6]: UX of the dropdown btn should be the same as for other btns

File: src/TitleDetails/AddPieceModal/ModalActionButtons/ModalActionButtons.js [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+      </Button>
+      <Dropdown
+        buttonProps={{
+          buttonStyle: 'primary',
+          buttonClass: css.dropdownButton,
+          marginBottom0: true,
+        }}
+      >
+        <DropdownMenu data-role="menu">
+          {actionMenus}
```
Comment (usavkov-epam) [inline src/TitleDetails/AddPieceModal/ModalActionButtons/ModalActionButtons.js:71]: why is it plural?

**Comment by usavkov-epam** [comment]
Please check both `rtl` and `ltr` directions:

image

**Review by usavkov-epam — APPROVED** [review]

**Pull request merged**